src/main/java/org/broadinstitute/consent/http/service/DarCollectionService.java

```diff
@@ -144,4 +155,30 @@ public DarCollection updateDarCollection(Integer collectionId, User user) {
     return collections;
   }
 
+  // fetch elections in bulk via dar reference ids
+  // if any election exists, throw a bad request
+  // else collect the dar reference ids where the status is NOT cancelled
+  // once collected, perform bulk update where status is updated with cancelled
+  public DarCollection cancelDarCollection(DarCollection collection, User user) {
+    List<DataAccessRequest> dars = collection.getDars();
+    List<String> referenceIds = dars.stream()
+      .map(d -> d.getReferenceId())
+      .collect(Collectors.toList());
+    
+    List<Election> elections = electionDAO.findLastElectionsByReferenceIdsAndType(referenceIds, ElectionType.DATA_ACCESS.getValue());
```

Comment:
same thing here, `referenceIds` should never be empty, but we should be safer about catching that case.

Reply:
Should an error be thrown here? Logically I can end the function by just returning collection, but the code is discovering bad data which the user should know about.

Reply:
I would log a warning instead of pulling that up to the client.